packages/polkadart_scale_codec/lib/src/util/functions.dart

```diff
@@ -139,21 +125,25 @@ bool checkSignedBigInt(BigInt val, int bitSize) {
     default:
       throw UnexpectedCaseException('Unexpected BitSize: $bitSize.');
   }
-  return checkBigInt(val, 'I', bitSize, min, max);
+
+  if (!compare(value, min, max)) {
+    throw InvalidSizeException(
+        'Invalid value at i$bitSize. $value is out of range $min - $max.');
```

Comment:
A good format for exception message is to provide what is expected to happen and what is actually happening
```suggestion
        'Expected value to be between  $min and $max, but got $value');
```